doc/adrs/209_decouple_command_palette_from_editing_context.adoc

```diff
@@ -0,0 +1,52 @@
+= ADR-209 - Decouple the command palette from the editing context
+
+== Context
+
+For now the command palette is only used in the project workbench, and is tightly coupled to the editing context and the selection.
+We want to reuse the command palette in other contexts, including contexts where there is no editing context nor selection (e.g. the `projects` page).
+
+== Decision
+
+Remove the `editingContextId` from the `Omnibox` and `OmniboxProvider` and add function parameters to retrieve and execute commands.
+Remove the `useSelection` hook from the `Omnibox`.
+
+The properties of the `Omnibox` will be changed as follows:
+
+[source, typescript]
+----
+export interface OmniboxProps {
+  open: boolean;
+  getCommands: (query: string) => void;
+  commandLoading: boolean;
+  commandData: GQLGetOmniboxCommandsQueryData | null;
+  executeCommand: (commandId: string) => void;
+  executeCommandData: GQLExecuteOmniboxCommandData | null;
+  onClose: () => void;
+}
+----
+
+These new parameters will be added in other classes/functions to ensure they are correctly forwarded to the Omnibox.
+Specifically, the `OmniboxProvider#openOmnibox` function will require these parameters to let client code open an command palette with the appropriate `getCommands` and `executeCommand` functions.
+
+Sirius Web will provide these functions for the command palette that is displayed in the project workbench.
+These functions will rely on existing hooks to get the editing context (e.g. `useCurrentProject()`) and the selection (`useSelection()`), and will perform the same operation as they do now.
+
+Currently, the command palette has 2 modes: _commands_ and _search_. 
+While the modes should be provided as parameters of the command palette, a first implementation will keep them hardcoded, leaving this refactoring as future work.
+This may imply to provide the following additional parameters to the Omnibox:
+
+[source, typescript]
+----
+getSearchResults: (query: string) => void;
+searchResultsLoading: boolean;
+searchResultsData: GQLGetOmniboxSearchResultsQueryData | null;
```

Comment:
shouldn't those be commands too? Because in the project page, clicking on a search result should not use `useSelection()` to select it in the workbench